ownCloud/Client/Actions/Scanner/ScanViewController.swift

```diff
@@ -0,0 +1,526 @@
+//
+//  ScanViewController.swift
+//  ownCloud
+//
+//  Created by Felix Schwarz on 30.08.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2019, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+import VisionKit
+import ImageIO
+import ownCloudSDK
+import ownCloudApp
+
+class ScanImageView : UIImageView {
+	var aspectHeight : CGFloat? {
+		didSet {
+			self.invalidateIntrinsicContentSize()
+		}
+	}
+
+	override var intrinsicContentSize: CGSize {
+		if let aspectHeight = aspectHeight, let image = image {
+			let imageSize = image.size
+			var intrinsicSize : CGSize = CGSize(
+				width: imageSize.width * aspectHeight / imageSize.height,
+				height: aspectHeight
+			)
+
+			if intrinsicSize.width < 1 { intrinsicSize.width = 1 }
+			if intrinsicSize.height < 1 { intrinsicSize.height = 1 }
+
+			return intrinsicSize
+		}
+
+		return super.intrinsicContentSize
+	}
+}
+
+class ScanPage {
+	var image : UIImage
+
+	init(with image: UIImage) {
+		self.image = image
+	}
+}
+
+class ScanPageCell : UICollectionViewCell, Themeable {
+	private var imageView : ScanImageView?
+
+	var page : ScanPage? {
+		didSet {
+			self.imageView?.image = page?.image
+			self.imageView?.invalidateIntrinsicContentSize()
+		}
+	}
+
+	var aspectHeight : CGFloat = 0 {
+		didSet {
+			if aspectHeight != 0 {
+				imageView?.aspectHeight = aspectHeight
+			}
+		}
+	}
+
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+
+		imageView = ScanImageView()
+		imageView?.translatesAutoresizingMaskIntoConstraints = false
+		imageView?.contentMode = .scaleAspectFill
+		imageView?.setContentCompressionResistancePriority(.required, for: .horizontal)
+		imageView?.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+
+		guard let imageView = imageView else { return }
+
+		self.contentView.addSubview(imageView)
+
+		NSLayoutConstraint.activate([
+			imageView.leftAnchor.constraint(equalTo: contentView.leftAnchor),
+			imageView.rightAnchor.constraint(equalTo: contentView.rightAnchor),
+			imageView.topAnchor.constraint(equalTo: contentView.topAnchor),
+			imageView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
+		])
+
+		self.layer.shadowColor = UIColor.black.cgColor
+		self.layer.shadowRadius = 6
```

Comment:
@felix-schwarz Consider converting magic numbers to constants?

Reply:
Done.